#! /bin/sh -e

if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
elif [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch) patch $pdir -f --no-backup-if-mismatch -p1 < $0;;
    -unpatch) patch $pdir -f --no-backup-if-mismatch -R -p1 < $0;;
    *)
	echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
	exit 1
esac
exit 0

# DP: Fix "suspend" segfaults in non-interactive shells

--- bash/builtins/common.c~	2002-06-28 18:24:31.000000000 +0200
+++ bash/builtins/common.c	2003-10-28 10:30:15.000000000 +0100
@@ -244,7 +244,7 @@
      char *s;
 {
   if (s)
-    builtin_error ("%s: no job control");
+    builtin_error ("%s: no job control", s);
   else
     builtin_error ("no job control");
 }
